models/test/mapper/TestPreviewMapper.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTestPreviewer\models\test\mapper;
+
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoQtiTestPreviewer\models\test\TestPreviewConfig;
+use oat\taoQtiTestPreviewer\models\test\TestPreviewMap;
+use qtism\data\AssessmentTest;
+use qtism\data\IAssessmentItem;
+use qtism\data\NavigationMode;
+use qtism\runtime\tests\Route;
+use qtism\runtime\tests\RouteItem;
+
+class TestPreviewMapper extends ConfigurableService implements TestPreviewMapperInterface
+{
+    use OntologyAwareTrait;
+
+    public function map(AssessmentTest $test, Route $route, TestPreviewConfig $config): TestPreviewMap
+    {
+        $map = [
+            'scope' => 'test',
+            'parts' => []
+        ];
+
+        $routeItems = $route->getAllRouteItems();
+        $checkInformational = $config->get(TestPreviewConfig::CHECK_INFORMATIONAL);
+        $forceInformationalTitles = $config->get(TestPreviewConfig::REVIEW_FORCE_INFORMATION_TITLE);
+        $displaySubsectionTitle = $config->get(TestPreviewConfig::REVIEW_DISPLAY_SUBSECTION_TITLE) === null
+            ? true
+            : $config->get(TestPreviewConfig::REVIEW_DISPLAY_SUBSECTION_TITLE);
+
+        $map['title'] = $test->getTitle();
+        $map['identifier'] = $test->getIdentifier();
+        $map['className'] = $test->getQtiClassName();
+        $map['toolName'] = $test->getToolName();
+        $map['exclusivelyLinear'] = $test->isExclusivelyLinear();
+        $map['hasTimeLimits'] = $test->hasTimeLimits();
+
+        $offset = 0;
+        $offsetSection = 0;
+        $lastSection = null;
+
+        /** @var RouteItem $routeItem */
+        foreach ($routeItems as $routeItem) {
+            $itemRefs = $this->getRouteItemAssessmentItemRefs($routeItem);
+
+            /** @var IAssessmentItem $itemRef */
+            foreach ($itemRefs as $itemRef) {
+                $occurrence = $routeItem->getOccurence();
+
+                $isItemInformational = true; //@TODO Implement this as a feature
```

Comment:
Is this @todo shall be solved before merge?

Reply:
There are some other TODOS intentionally left, cause we will implement those features in future test-previewer versions. This is the simpler version.

Reply:
Do we have some tickets (planned?) to tackle that?

Reply:
On demand as we make the test preview work as the real test runner we will cover them, but as far as I know, no tasks :/